### PR TITLE
Add withOccurrences and withoutOccurrences to MutableBag.

### PR DESCRIPTION
**Pull request by donraab**

### CODE REVIEW

File: eclipse-collections-api/src/main/java/org/eclipse/collections/api/bag/MultiReaderBag.java [hunk 1]
```diff
@@ -41,6 +41,20 @@ default MultiReaderBag<T> without(T element)
         return this;
     }
 
+    @Override
+    default MultiReaderBag<T> withOccurrences(T element, int occurrences)
+    {
+        this.addOccurrences(element, occurrences);
```
Comment (nikhilnanivadekar) [inline eclipse-collections-api/src/main/java/org/eclipse/collections/api/bag/MultiReaderBag.java:47]: Just to confirm, we use this pattern for Multi reader collections correct? The return this is why I am asking

File: unit-tests/src/test/java/org/eclipse/collections/impl/bag/mutable/MutableBagTestCase.java [hunk 2]
```diff
@@ -278,12 +278,31 @@ public void addOccurrences()
         MutableBagTestCase.assertBagsEqual(HashBag.newBagWith("1", "2", "2", "2", "2", "2", "2", "3"), bag);
     }
 
+    @Test
```
Comment (nikhilnanivadekar) [inline unit-tests/src/test/java/org/eclipse/collections/impl/bag/mutable/MutableBagTestCase.java:281]: Do we need to assert the instance is the same as well?

File: eclipse-collections-api/src/main/java/org/eclipse/collections/api/bag/MultiReaderBag.java (hunk 1 above)
Comment (donraab) [reply eclipse-collections-api/src/main/java/org/eclipse/collections/api/bag/MultiReaderBag.java:47]: Yes, the `addOccurrences` method will take a lock, so I didn't introduce the lock in this override.

**Review by nikhilnanivadekar — APPROVED** [review]

File: unit-tests/src/test/java/org/eclipse/collections/impl/bag/mutable/MutableBagTestCase.java (hunk 2 above)
Comment (donraab) [reply unit-tests/src/test/java/org/eclipse/collections/impl/bag/mutable/MutableBagTestCase.java:281]: Yes, that is a good thing to assert as well. I can add it.

**Pull request merged**